indico/modules/attachments/clone.py

```diff
@@ -16,56 +16,78 @@
 
 from __future__ import unicode_literals
 
+from sqlalchemy.orm import subqueryload, joinedload
+
 from indico.core.db import db
+from indico.core.db.sqlalchemy.links import LinkType
 from indico.core.db.sqlalchemy.util.models import get_simple_column_attrs
 from indico.modules.attachments.models.attachments import Attachment, AttachmentFile, AttachmentType
 from indico.modules.attachments.models.folders import AttachmentFolder
+from indico.modules.events.cloning import EventCloner
 from indico.util.i18n import _
-from MaKaC.conference import EventCloner
 
 
 class AttachmentCloner(EventCloner):
-    def find_attachments(self):
-        return Attachment.find(~AttachmentFolder.is_deleted, ~Attachment.is_deleted,
-                               AttachmentFolder.event_id == int(self.event.id), _join=AttachmentFolder)
+    name = 'attachments'
+    friendly_name = _('Materials')
+    uses = {'sessions', 'contributions'}
+
+    @property
+    def is_available(self):
+        return bool(self.old_event.all_attachment_folders.filter(AttachmentFolder.attachments.any()).count())
 
-    def find_folders(self):
-        return self.event.as_event.attachment_folders.filter_by(is_deleted=False)
+    def run(self, new_event, cloners, shared_data):
+        self._clone_nested_attachments = False
+        self._session_map = self._contrib_map = self._subcontrib_map = None
+        if cloners >= {'sessions', 'contributions'}:
+            self._clone_nested_attachments = True
+            self._session_map = shared_data['sessions']['session_map']
+            self._contrib_map = shared_data['contributions']['contrib_map']
+            self._subcontrib_map = shared_data['contributions']['subcontrib_map']
+        with db.session.no_autoflush:
+            self._clone_attachments(new_event)
+        db.session.flush()
 
-    def get_options(self):
-        enabled = bool(self.find_attachments().count())
-        return {'attachments': (_("Materials"), enabled, False)}
+    def _query_folders(self, base_query, for_event):
+        query = (base_query
+                 .filter_by(is_deleted=False)
```

Comment:
i know, however i remember you commenting on something similar :P (that we should place `filter/filter_by` at the end)